contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -472,6 +473,30 @@ contract AerodromeAMOStrategy is InitializableAbstractStrategy {
         }
     }
 
+    /**
+     * @dev Remove just enough liquidity so that the swap operation can be performed. No need to
+     * add the gaugeUnstakeAndRestake modifier since the underlying _removeLiquidity function
+     * will do that.
+     * Good to consider that remaining liquidity will remain in the pool & staked to the gauge.
+     *
+     * @param _amountToSwap The amount of the token to swap
+     * @param _swapWeth Swap using WETH when true, use OETHb when false
+     */
+    function _removeLiquidityToEnsureSwap(uint256 _amountToSwap, bool _swapWeth)
+        internal
+    {
+        // swapping OETHb to WETH doesn't require liquidity removal or when there
+        // is no amount to be swapped
+        if (!_swapWeth || _amountToSwap == 0) {
+            return;
+        }
+
+        _ensureWETHBalance(_amountToSwap);
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
```

Comment:
nit: This placement of burn is a bit off but it seems to work fine.

If `swapAmount < depositAmount`, there'd be no burn here. So that's good. If `depositAmount < swapAmount`, it pulls WETH from the pool and burns superOETHb. But there'll be nothing to add post that swap. Code seems to handle both cases fine. Will do a few checks to be sure and get back on this

Reply:
@sparrowDom If I just call rebalance with swapAmount set to 10 WETH and deposit amount set to 0 WETH (i.e. no WETH from Vault), this is how the strategy behaves (assuming 20/80 splits):
- Remove 10 WETH and 40 superOETHb
- Burn 40 superOETHb
- Swap 10 WETH for 40 superOETHb
- Exit without deposit (since depositAmount is 0)

So, it looks like it still leaves some superOETHb in the contract

Reply:
Perhaps we could consider updating `_solvencyAssert` to ensure that `oethb.balanceOf(address(this)) == 0` (or even some small amount like 1e6)

Reply:
this is a great point. Will add a test for this. Thinking that I'd rather burn the OETH within the swap function so that each function in the `_rebase` flow cleans up after itself. The sanity check in the end also makes sense

Reply:
Yeah. We are on L2's with this, so gas is cheap.

Reply:
> since someone could trigger it by donating?

I guess we can always burn any leftover superOETHb at the end of a rebalance (just before calling the solvency check method)

Reply:
I was curious why [the new test conditions](https://github.com/OriginProtocol/origin-dollar/pull/2276/files#diff-2fc9b69b9f62ab0307369e83cceb6e19bdf34994b331eb4f5e21a8fbf9341cefR1222-R1231) I have added before this conversation didn't trigger OETHb being left on the contract. 
Turns out that because we slightly overshoot the amount of WETH being removed to ensure the swap the `addLiquidity` function would always execute in full - adding the liquidity and burning OETHb.

It is somewhat comforting that when (in future) we would fix the slight overshooting of WETH in rebalance those test would fail. Still super awesome this was also caught in the code review 💪